### PR TITLE
Fix generating package doc CFI paths with XML namespaces

### PR DESCRIPTION
**Pull request by JCCR**
#### Related issue(s) and/or pull request(s)
#7 

#### Test cases, sample files
Updated the test suite to help in this case.

### CODE REVIEW

File: js/cfi_generator.js [hunk 1]
```diff
@@ -193,10 +193,12 @@ var obj = {
         this.validatePackageDocument(packageDocument, contentDocumentName);
 
         // Get the start node (itemref element) that references the content document
-        $itemRefStartNode = $("itemref[idref='" + contentDocumentName + "']", $(packageDocument));
+        var $itemRefStartNode = $($(packageDocument.getElementsByTagNameNS('*', 'itemref')).toArray().find(function (node) {
+            return node.getAttribute('idref') === contentDocumentName;
+        }));
```
Comment (danielweck) [inline js/cfi_generator.js]: Support for `array.find()` in web browsers is not broad enough, it seems:
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/find

File: js/cfi_interpreter.js [hunk 2]
```diff
@@ -55,7 +55,6 @@ var obj = {
     //   the reading system, as it stands now.
     getContentDocHref : function (CFI, packageDocument, classBlacklist, elementBlacklist, idBlacklist) {
 
-        var $packageDocument = $(packageDocument);
         var decodedCFI = decodeURI(CFI);
```
Comment (danielweck) [inline js/cfi_interpreter.js]: The `packageDocument` function parameter is not a jQuery object anymore in `searchLocalPathForHref`, `getContentDocHref `, and `retrieveItemRefHref `. This is good, but this is also a major API change, which could break existing consumer / client code. Any thoughts?

File: js/cfi_generator.js (hunk 1 above)
Comment (danielweck) [inline js/cfi_generator.js]: Note that jQuery's `toArray()` return a standard ECMAScript array: https://api.jquery.com/toArray/

**Comment by danielweck** [comment]
Related: https://github.com/readium/readium-cfi-js/pull/33

File: js/cfi_interpreter.js (hunk 2 above)
Comment (danielweck) [reply js/cfi_interpreter.js]: Follow-up: I suppose $(obj) === obj when obj is already a $ jQuery object, right? So this "normalisation" step allows for both vanilla JS objects (DOM éléments) and jQuery objects to be used as function parameter ... so I retract my comment.

File: js/cfi_generator.js (hunk 1 above)
Comment (danielweck) [reply js/cfi_generator.js]: IE does not have support at all. can we use underscore or jQuery find() instead?

**Comment by JCCR** [comment]
@danielweck 
I took your feedback into account in my latest commit to this.

**Comment by danielweck** [comment]
I have not tested the code, but I read the unit tests, and the I reviewed the code changes. Looks good to me.

**Pull request merged**